### PR TITLE
typescript-node: handle additional properties

### PR DESCRIPTION
**Pull request by mosesonline**
Fixes: #4698

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.
@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @nicokoenig (2018/09) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11)

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptNodeClientCodegen.java [hunk 1]
```diff
@@ -289,14 +291,23 @@ private String getPackageRootDirectory() {
...
         String name = classname.substring(0, classname.length() - apiSuffix.length());
         return toApiFilename(name);
     }
-    
+
     private String getModelnameFromModelFilename(String filename) {
         String name = filename.substring((modelPackage() + File.separator).length());
         return camelize(name);
     }
+
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptNodeClientCodegen.java]:
```suggestion
@Override
```

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptNodeClientCodegen.java [hunk 2]
```diff
@@ -289,14 +291,23 @@ private String getPackageRootDirectory() {
...
+    protected void addAdditionPropertiesToCodeGenModel(CodegenModel codegenModel, Schema schema) {
+        Schema additionalProperties = ModelUtils.getAdditionalProperties(schema);
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptNodeClientCodegen.java]:
```suggestion
        super.addAdditionPropertiesToCodeGenModel(codegenModel, schema);
        Schema additionalProperties = ModelUtils.getAdditionalProperties(schema);
```

**Comment by macjohnny** [comment]
@mosesonline thanks for your PR!

**Comment by mosesonline** [comment]
> @mosesonline thanks for your PR!

Thank you and the other contributors for the work. Glad to help.

**Pull request merged**